wire-ios-data-model/Source/Model/Conversation/ZMConversation+MLS.swift

```diff
@@ -133,6 +136,37 @@ extension ZMConversation {
     /// to commit.
     @NSManaged public var commitPendingProposalDate: Date?
 
+    @NSManaged private var primitiveMlsVerificationStatus: NSNumber?
+
+    /// The mls verification status.
+    ///
+    /// If this conversation is an mls group (which it should be if the
+    /// `messageProtocol` is `mls`), then this identifier should exist.
+
+    public var mlsVerificationStatus: MLSVerificationStatus? {
+        get {
+            willAccessValue(forKey: Self.mlsVerificationStatusKey)
+            let value = primitiveMlsVerificationStatus?.int16Value
+            didAccessValue(forKey: Self.mlsVerificationStatusKey)
+
+            guard let value = value else {
+                return nil
+            }
+
+            guard let status = MLSVerificationStatus(rawValue: value) else {
+                fatalError("failed to init MLSVerificationStatus from rawValue: \(value)")
+            }
+
+            return status
+        }
+
+        set {
+            willChangeValue(forKey: Self.mlsVerificationStatusKey)
+            primitiveMlsVerificationStatus = newValue.map { NSNumber(value: $0.rawValue) }
+            didChangeValue(forKey: Self.mlsVerificationStatusKey)
+        }
+    }
```

Comment:
I wonder if this property could be non-optional with the raw value of `.notVerified`.

```suggestion
    @NSManaged private var primitiveMlsVerificationStatus: NSNumber

    /// The mls verification status.
    ///
    /// If this conversation is an mls group (which it should be if the
    /// `messageProtocol` is `mls`), then this identifier should exist.

    public var mlsVerificationStatus: MLSVerificationStatus {
        get {
            willAccessValue(forKey: Self.mlsVerificationStatusKey)
            let value = primitiveMlsVerificationStatus.int16Value
            didAccessValue(forKey: Self.mlsVerificationStatusKey)

            guard let status = MLSVerificationStatus(rawValue: value) else {
                fatalError("failed to init MLSVerificationStatus from rawValue: \(value)")
            }

            return status
        }
        set {
            willChangeValue(forKey: Self.mlsVerificationStatusKey)
            primitiveMlsVerificationStatus = .init(value: newValue.rawValue)
            didChangeValue(forKey: Self.mlsVerificationStatusKey)
        }
    }
```

Reply:
I'm fine to have this property as non-optional. For `e2ei` this shouldn't break anything. @johnxnguyen What do you think? I know we have some other mls properties that are optional.